stdlib/option.mli

```diff
@@ -0,0 +1,75 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Option values.
+
+    @since NEXT_RELEASE *)
+
+(** {1:options Options} *)
+
+type 'a t = 'a option
+(** The type for option values. Either [Some v] or [None]. *)
+
+val some : 'a -> 'a option
+(** [some v] is [Some v]. *)
+
+val evict : none:'a -> 'a option -> 'a
+(** [evict ~none o] is [none] if [o] is [None] and [v] if [o] is [Some v]. *)
+
+val get : 'a option -> 'a
+(** [get o] is [v] if [o] is [Some v] and @raise Invalid_argument otherwise. *)
+
+val map : ('a -> 'b) -> 'a option -> 'b option
+(** [map f o] is [None] if [o] is [None] and [Some (f v)] is [o] is [Some v]. *)
+
+val flatten : 'a option option -> 'a option
+(** [flatten oo] is [Some v] if [oo] is [Some (Some v)] and [None]
+    otherwise. *)
+
+val iter : ('a -> unit) -> 'a option -> unit
+(** [iter f o] is [f v] if [o] is [Some v] and [()] otherwise. *)
+
+val fold : ('a -> 'b -> 'b) -> 'a option -> 'b -> 'b
+(** [fold f o acc] is [f v acc] if [o] is [Some v] and [acc] otherwise. *)
+
+val hash : ('a -> int) -> 'a option -> int
+(** [hash h o] hashes [o] with the help of [h] on [Some _] values. *)
+
+(** {1:preds Predicates and comparisons} *)
+
+val is_none : 'a option -> bool
```

Comment:
It's completely minor, but to be coherent with `of_result`, you should either use `_` for all ignored type variables (`val is_none : _ option -> bool`) or none of them (`val of_result : ('a, 'b) result -> 'a option`).  I believe using `_` might be slightly confusing for beginner, so I'd have a preference for introducing (useless) variable names.

Reply:
Done (named it so that comment won't outdate itself).